assets/app/reducers.js

```diff
@@ -39,36 +39,21 @@ export function builds(state = [], action) {
 export function sites(state = [], action) {
   switch (action.type) {
     case siteActionTypes.SITES_RECEIVED:
-      return action.sites || []
+      return action.sites || [];
     case siteActionTypes.SITE_ADDED:
       // if a site hasn't been properly added, return the existing state
       // TODO: why is this getting called if there is an error
       return action.site ? [...state, action.site] : state;
     case siteActionTypes.SITE_UPDATED:
-      return state.map((site) => {
-        if (site.id !== action.siteId) return site;
-
-        return Object.assign({}, site, action.site);
-      });
-    case siteActionTypes.SITE_DELETED:
-      return state.filter((site) => site.id != action.siteId);
+      return mapPropertyToMatchingSite(state, action.siteId, action.site);
     case siteActionTypes.SITE_CONFIGS_RECEIVED:
-      return state.map((s) => {
-        if (s.id !== action.siteId) return s;
-        return Object.assign({}, s, action.configs);
-      });
+      return mapPropertyToMatchingSite(state, action.siteId, action.configs);
     case siteActionTypes.SITE_ASSETS_RECEIVED:
-      return state.map((s) => {
-        if (s.id !== action.siteId) return s;
-        return Object.assign({}, s, { assets: action.assets });
-      });
+      return mapPropertyToMatchingSite(state, action.siteId, { assets: action.assets });
     case siteActionTypes.SITE_CONTENTS_RECEIVED:
-      return state.map((s) => {
-        if (s.id !== action.siteId) return s;
-        return Object.assign({}, s, {
-          files: action.files
-        });
-      });
+      return mapPropertyToMatchingSite(state, action.siteId, { files: action.files });
+    case siteActionTypes.SITE_DELETED:
+      return state.filter((site) => site.id != action.siteId);
     case siteActionTypes.SITE_CHILD_CONTENT_RECEIVED:
       const currentMap = state.childDirectoriesMap || {};
```

Comment:
ahh I see. Yes, that is not correct, I think its a bad copy/paste from some previous code, when I put the child contents in a separate reducer (for some reason).

Most of that work should get moved into the function that is applied to the list of sites currently in the state array.